ironfish/src/rpc/routes/wallet/createTransaction.ts

```diff
@@ -0,0 +1,250 @@
+      if (receive.assetId) {
+        assetId = Buffer.from(receive.assetId, 'hex')
+      }
+
+      const amount = CurrencyUtils.decode(receive.amount)
+      if (amount <= 0) {
+        throw new ValidationError(`Invalid transaction amount ${amount}.`)
+      }
+
+      const sum = totalByAssetIdentifier.get(assetId) ?? 0n
+      totalByAssetIdentifier.set(assetId, sum + amount)
+
+      return {
+        publicAddress: receive.publicAddress,
+        amount: amount,
+        memo: receive.memo,
+        assetId,
+      }
+    })
+
+    const mints: MintDescription[] = []
+    if (data.mints) {
+      for (const mint of data.mints) {
+        let asset: Asset
+        if (mint.assetId) {
+          const record = await node.chain.getAssetById(Buffer.from(mint.assetId, 'hex'))
+          if (!record) {
+            throw new ValidationError(
+              `Asset not found. Cannot mint for identifier '${mint.assetId}'`,
+            )
+          }
+
+          asset = new Asset(
+            account.spendingKey,
+            record.name.toString('utf8'),
+            record.metadata.toString('utf8'),
+          )
+          // Verify the stored asset produces the same identfier before building a transaction
+          if (!asset.id().equals(Buffer.from(mint.assetId, 'hex'))) {
+            throw new ValidationError(`Unauthorized to mint for asset '${mint.assetId}'`)
+          }
+        } else {
+          if (mint.name === undefined) {
+            throw new ValidationError('Must provide name or identifier to mint')
+          }
+          asset = new Asset(account.spendingKey, mint.name, mint.metadata ?? '')
+        }
+
+        mints.push({
+          asset: asset,
+          value: CurrencyUtils.decode(mint.value),
+        })
+      }
+    }
+
+    const burns: BurnDescription[] = []
+    if (data.burns) {
+      data.burns.map((burn) => {
+        let assetId = Asset.nativeId()
+        if (burn.assetId) {
+          assetId = Buffer.from(burn.assetId, 'hex')
+        }
+        burns.push({
+          assetId: assetId,
+          value: CurrencyUtils.decode(burn.value),
+        })
+      })
+    }
+
+    // Check that the node account is updated
+    for (const [assetId, sum] of totalByAssetIdentifier) {
+      const balance = await node.wallet.getBalance(account, assetId)
+
+      if (balance.confirmed < sum) {
+        throw new ValidationError(
+          `Your balance is too low. Add funds to your account first`,
+          undefined,
+          ERROR_CODES.INSUFFICIENT_BALANCE,
+        )
+      }
+    }
+
+    let rawTransaction
+
+    try {
+      if (data.fee) {
+        rawTransaction = await node.wallet.createTransaction(account, receives, mints, burns, {
+          fee: BigInt(data.fee),
+          expirationDelta:
+            data.expirationDelta ?? node.config.get('transactionExpirationDelta'),
+          expiration: data.expiration,
+        })
+      } else {
+        const feeRate = BigInt(
```

Comment:
CurrencyUtils.decode(data.feeRate), fee rates are currency utils with a unit of `ore / 1 KB`